flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java

```diff
@@ -222,14 +223,23 @@ protected void registerNamespace(
                             Collections.singletonList(simplifiedRexNode),
                             reducedNodes);
             // check whether period is the unsupported expression
-            if (!(reducedNodes.get(0) instanceof RexLiteral)) {
-                throw new UnsupportedOperationException(
+            final RexNode reducedNode = reducedNodes.get(0);
+            if (!(reducedNode instanceof RexLiteral)) {
+                throw new ValidationException(
                         String.format(
                                 "Unsupported time travel expression: %s for the expression can not be reduced to a constant by Flink.",
                                 periodNode));
             }
 
-            RexLiteral rexLiteral = (RexLiteral) (reducedNodes).get(0);
+            RexLiteral rexLiteral = (RexLiteral) reducedNode;
+            final SqlTypeName sqlTypeName = rexLiteral.getTypeName();
+            if (!(sqlTypeName == SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE
+                    || sqlTypeName == SqlTypeName.TIMESTAMP)) {
```

Comment:
```suggestion
            if (!SqlTypeUtil.isTimestamp(sqlTypeName) {
```
I think `org.apache.calcite.sql.type.SqlTypeUtil#isTimestamp` could be reused here
as defined at 
https://github.com/apache/calcite/blob/413eded693a9087402cc1a6eefeca7a29445d337/core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java#L390-L393

and 
https://github.com/apache/calcite/blob/413eded693a9087402cc1a6eefeca7a29445d337/core/src/main/java/org/apache/calcite/sql/type/SqlTypeFamily.java#L182-L183

Reply:
Good idea! I was looking for sth like it, but I failed.